controllers/kserve_inferenceservice_controller_test.go

```diff
@@ -94,6 +103,119 @@ var _ = Describe("The Openshift Kserve model controller", func() {
+
+					return fmt.Errorf("Annotation[constants.ServingCertAnnotationKey] is not added yet")
+				}
+				return nil
+			}, timeout, interval).Should(Succeed())
+
+			Expect(isvcService.Annotations[constants.ServingCertAnnotationKey]).Should(Equal(inferenceService.Name))
+		})
+
+		It("should create a secret for runtime and update kserve local gateway in the istio-system namespace", func() {
+			// We need to stub the cluster state and indicate where is istio namespace (reusing authConfig test data)
+			if dsciErr := createDSCI(DSCIWithoutAuthorization); dsciErr != nil && !errors.IsAlreadyExists(dsciErr) {
+				Fail(dsciErr.Error())
+			}
+			// Stub: Create a kserve-local-gateway, which must be created by the OpenDataHub operator.
+			kserveLocalGateway := &istioclientv1beta1.Gateway{}
+			err := convertToStructuredResource(kserveLocalGatewayPath, kserveLocalGateway)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cli.Create(ctx, kserveLocalGateway)).Should(Succeed())
+
+			// Stub: Create a certificate Secret, which must be created by the openshift service-ca operator.
+			secret := &corev1.Secret{}
+			err = convertToStructuredResource(testIsvcSvcSecretPath, secret)
+			Expect(err).NotTo(HaveOccurred())
+			secret.SetNamespace(testNs)
+			Expect(cli.Create(ctx, secret)).Should(Succeed())
+
+			// Create a new InferenceService
+			inferenceService := &kservev1beta1.InferenceService{}
+			err = convertToStructuredResource(KserveInferenceServicePath1, inferenceService)
+			Expect(err).NotTo(HaveOccurred())
+			inferenceService.SetNamespace(testNs)
+
+			Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+
+			// Update the URL of the InferenceService to indicate it is ready.
+			deployedInferenceService := &kservev1beta1.InferenceService{}
+			err = cli.Get(ctx, types.NamespacedName{Name: inferenceService.Name, Namespace: inferenceService.Namespace}, deployedInferenceService)
+			Expect(err).NotTo(HaveOccurred())
+
+			newAddress := &duckv1.Addressable{
+				URL: apis.HTTPS("example-onnx-mnist-default.test.com"),
+			}
+			deployedInferenceService.Status.Address = newAddress
+
+			err = cli.Status().Update(ctx, deployedInferenceService)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Verify that the certificate secret is created in the istio-system namespace.
+			Eventually(func() error {
+				secret := &corev1.Secret{}
+				err := cli.Get(ctx, client.ObjectKey{Namespace: constants.IstioNamespace, Name: fmt.Sprintf("%s-%s", inferenceService.Name, inferenceService.Namespace)}, secret)
+				if err != nil {
+					return err
+				}
+				return nil
+			}, timeout, interval).Should(Succeed())
+
+			// Verify that the gateway is updated in the istio-system namespace.
+			var gateway *istioclientv1beta1.Gateway
+			Eventually(func() error {
+				gateway, err = waitForUpdatedGatewayCompletion(cli, "add", constants.IstioNamespace, constants.KServeGatewayName, inferenceService.Name)
+				if err != nil {
+					return err
+				}
+				return nil
```

Comment:
```suggestion
				return err
```

There are several ones like this. I won't point all them :-)

Reply:
I updated all similar cases.